src/main/java/org/mockito/internal/util/reflection/InstrumentationMemberAccessor.java

```diff
@@ -0,0 +1,363 @@
+                                    ClassLoadingStrategy.Default.WRAPPER)
+                            .getLoaded()
+                            .getConstructor()
+                            .newInstance();
+            throwable = null;
+        } catch (Throwable t) {
+            instrumentation = null;
+            dispatcher = null;
+            throwable = t;
+        }
+        INSTRUMENTATION = instrumentation;
+        DISPATCHER = dispatcher;
+        INITIALIZATION_ERROR = throwable;
+    }
+
+    private final MethodHandle getModule, isOpen, redefineModule, privateLookupIn;
+
+    InstrumentationMemberAccessor() {
+        if (INITIALIZATION_ERROR != null) {
+            throw new MockitoInitializationException(
+                    join(
+                            "Could not initialize the Mockito instrumentation member accessor",
+                            "",
+                            "This is unexpected on JVMs from Java 9 or later - possibly, the instrumentation API could not be resolved"),
+                    INITIALIZATION_ERROR);
+        }
+        try {
+            Class<?> module = Class.forName("java.lang.Module");
+            getModule =
+                    MethodHandles.publicLookup()
+                            .findVirtual(Class.class, "getModule", MethodType.methodType(module));
+            isOpen =
+                    MethodHandles.publicLookup()
+                            .findVirtual(
+                                    module,
+                                    "isOpen",
+                                    MethodType.methodType(boolean.class, String.class, module));
+            redefineModule =
+                    MethodHandles.publicLookup()
+                            .findVirtual(
+                                    Instrumentation.class,
+                                    "redefineModule",
+                                    MethodType.methodType(
+                                            void.class,
+                                            module,
+                                            Set.class,
+                                            Map.class,
+                                            Map.class,
+                                            Set.class,
+                                            Map.class));
+            privateLookupIn =
+                    MethodHandles.publicLookup()
+                            .findStatic(
+                                    MethodHandles.class,
+                                    "privateLookupIn",
+                                    MethodType.methodType(
+                                            MethodHandles.Lookup.class,
+                                            Class.class,
+                                            MethodHandles.Lookup.class));
+        } catch (Throwable t) {
+            throw new MockitoInitializationException(
```

Comment:
If this exception is thrown, should we automatically fall back to the reflection accessor? Or should we bail out at all times?

Reply:
This would mean that Java 9 does not contain a method that is speced to be there. I think an exception is appropriate. The use is guarded by `ModuleMemberAccess`.